components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/util/CibaAuthUtil.java

```diff
@@ -0,0 +1,362 @@
+
+package org.wso2.carbon.identity.oauth.ciba.util;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth.ciba.common.AuthenticationStatus;
+import org.wso2.carbon.identity.oauth.ciba.common.CibaParams;
+import org.wso2.carbon.identity.oauth.ciba.dto.AuthzRequestDTO;
+import org.wso2.carbon.identity.oauth.ciba.dto.CibaAuthRequestDTO;
+import org.wso2.carbon.identity.oauth.ciba.dto.CibaAuthResponseDTO;
+import org.wso2.carbon.identity.oauth.ciba.exceptions.CibaCoreException;
+import org.wso2.carbon.identity.oauth.ciba.exceptions.ErrorCodes;
+import org.wso2.carbon.identity.oauth.ciba.internal.CibaServiceDataHolder;
+import org.wso2.carbon.identity.oauth.ciba.model.CibaAuthCodeDO;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Provides utilities for the functioning of other classes.
+ */
+
+public class CibaAuthUtil {
+
+    private static final Log log = LogFactory.getLog(CibaAuthUtil.class);
+
+    private CibaAuthUtil() {
+
+    }
+
+    private static CibaAuthUtil cibaAuthUtilInstance = new CibaAuthUtil();
+
+    public static CibaAuthUtil getInstance() {
```

Comment:
No need for singleton for the util class, we can make the methods util and use

Reply:
Do you mean static methods?

Reply:
Refactored.